### ISSUE TITLE
DateTime serialization and deserialization timezone

### ISSUE THREAD

**Issue by WongKit** [msgpack/msgpack-cli#347]
Hi,

I think there is an issue with DateTime serialization regarding timezones. I am located in Germany (timezone MESZ/ UTC+2 atm). When I serialize and deserialize a DateTime object, the date is not the same.

```
MessagePackSerializer mps = MessagePackSerializer.Get<DateTime>();
DateTime from = DateTime.Now;
DateTime to = (DateTime)mps.FromMessagePackObject(mps.ToMessagePackObject(from));

Console.WriteLine(from); //outputs 18:36:27 (correct time, MESZ)
Console.WriteLine(to); //outputs **16:36:27** (utc 0)
```

The problem does not occur, if DefaultDateTimeConversionMethod is set to DateTimeConversionMethod.Native. However, I still would expect the same DateTime if (de-)serialized from the same context.

Thanks and best regards
WongKit

**Comment by yfakariya** [comment]
First, in fact, msgpack serializes date-time value in Unix epoc (old spec) or timestamp format (new spec). Both of them uses UTC and do not carry any timezone related information. `DateTimeConversionMethod.Native` means uses "native representation of .NET" to reserve timezone or kind related information at the expense of interoperability. So, it is by design.

But I think that I do not understand what is your matter. If you intended a change request rather than a question, tell me your expectation (desired behavior of msgpack for you).

Thanks!

**Comment by WongKit** [comment]
Thank you for your reply. I did not realize, that DateTime objects also contain some form of timezone information. So `from` and `to` in my upper example actually have the same time, if DateTimeKind is considered.

If MsgPack does not store this information, I can understand, why it defaults to UTC time. I just got confused, as the console outputs two different values before and after serialization.

Thanks and sorry for the false alarm!
